dds_web/api/schemas/project_schemas.py

```diff
@@ -84,3 +86,120 @@ def return_items(self, data, **kwargs):
         """Return project object."""
 
         return data.get("project_row")
+
+
+class ProjectContentSchema(ProjectRequiredSchema):
+    """ """
+
+    requested_items = marshmallow.fields.List(marshmallow.fields.String, required=False)
+    url = marshmallow.fields.Boolean(required=False, default=False)
+    get_all = marshmallow.fields.Boolean(required=False, default=False)
+
+    def find_contents(self, project, contents):
+
+        # All contents
+        all_contents_query = models.File.query.filter(
+            models.File.project_id == sqlalchemy.func.binary(project.id)
+        )
+
+        # Get all files
+        files = all_contents_query.filter(models.File.name.in_(contents)).all()
+
+        # Get not found paths - may be folders
+        new_paths = set(contents).difference(x.name for x in files)
+
+        # Get all folder contents
+        folder_contents = {
+            x: all_contents_query.filter(models.File.subpath.like(f"{x.rstrip(os.sep)}%")).all()
+            for x in new_paths
+        }
+
+        # Not found
+        not_found = {x: folder_contents.pop(x) for x, y in list(folder_contents.items()) if not y}
+
+        return files, folder_contents, not_found
+
+    @marshmallow.post_dump
+    def return_items(self, data, **kwargs):
+        """Return project contents as serialized."""
+
```

Comment:
Personally I would move almost everything in this Schema back into the get method of the files endpoint. I think of the schemas as a place to validate parameters and possibly check user permissions. How do you think of them? The decision where to put the logic in the end is of course yours though.

Reply:
That would mean that we would have very similar/identical code for `FileInfo` and `FileInfoAll`, and I was thinking that this schema could also be used in `MatchFiles` later on since we can specify which columns should be returned etc.

Reply:
But Schemas are also good for serializing and deserializing objects

Reply:
Code reusability is a good point. I guess I haven't gotten used to the schemas yet then.